### PR TITLE
Fix(FASTQ cleaning) Allow FASTQ file to not exist when unlinking

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
`cg compress clean fastq` is broken if the FASTQ files are included. Each file is being unlinked twice, but crashes the second time since it no longer exists.

### Changed

- Allows FASTQ files to not exist when unlinking

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/compress/compress.py [hunk 1]
```diff
@@ -286,8 +286,7 @@ def remove_fastq(self, fastq_first: Path, fastq_second: Path):
...
         if self.dry_run:
             return
-        fastq_first.unlink()
-        LOG.debug(f"First FASTQ in pair {fastq_first} removed")
-        fastq_second.unlink()
-        LOG.debug(f"Second FASTQ in pair {fastq_second} removed")
-        LOG.info("FASTQ files removed")
+        for fastq_file in [fastq_first, fastq_second]:
+            if fastq_file.exists():
+                fastq_first.unlink()
```
Comment (diitaz93) [inline cg/meta/compress/compress.py:291]: I think this is what you meant?
```suggestion
                fastq_file.unlink()
```

File: cg/meta/compress/compress.py [hunk 2]
```diff
@@ -286,8 +286,7 @@ def remove_fastq(self, fastq_first: Path, fastq_second: Path):
...
+                LOG.debug(f"First FASTQ in pair {fastq_file} removed")
```
Comment (diitaz93) [inline cg/meta/compress/compress.py:292]: I think this is what you meant?
```suggestion
                fastq_file.unlink()
                LOG.debug(f"FASTQ file {fastq_file} in pair removed")
```

**Review by diitaz93 — COMMENTED** [review]
Suggested some changes based on my interpretation of the function

File: cg/meta/compress/compress.py (hunk 2 above)
Comment (Vince-janv) [reply cg/meta/compress/compress.py:292]: Yes, I the old text is confusing 👍

File: tests/meta/compress/test_clean_fastq.py
Comment (diitaz93) [inline tests/meta/compress/test_clean_fastq.py:1]: It would also be great to have a test when only one of the paths exists

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

**Review by ChrOertlin — APPROVED** [review]
Looks good, some refactoring suggestions

File: cg/meta/compress/compress.py [hunk 3]
```diff
@@ -286,8 +286,7 @@ def remove_fastq(self, fastq_first: Path, fastq_second: Path):
         LOG.info(f"Will remove {fastq_first} and {fastq_second}")
         if self.dry_run:
             return
-        fastq_first.unlink()
-        LOG.debug(f"First FASTQ in pair {fastq_first} removed")
-        fastq_second.unlink()
-        LOG.debug(f"Second FASTQ in pair {fastq_second} removed")
-        LOG.info("FASTQ files removed")
+        for fastq_file in [fastq_first, fastq_second]:
```
[the remark this answers is not in the document]
Comment (Vince-janv) [reply cg/meta/compress/compress.py:289]: Too big of change in this PR

File: tests/meta/compress/test_clean_fastq.py [hunk 4]
```diff
@@ -42,7 +42,8 @@ def test_remove_fastqs(
     assert compression_object.spring_metadata_path.exists()
     expected_output: str = f"Will remove {fastq_first} and {fastq_second}"
     assert expected_output in caplog.text
-    assert "FASTQ files removed" in caplog.text
+    assert f"FASTQ file {fastq_first} removed" in caplog.text
```
[the remark this answers is not in the document]
Comment (Vince-janv) [reply tests/meta/compress/test_clean_fastq.py:45]: Since the structure is in pairs, I think it makes sense to work that way for now

**Pull request merged**